### PR TITLE
feat: token balances across networks -- footer view for token details page

### PR DESCRIPTION
**Pull request by kayleegeorge**
Add network balances footer view component for token details page when <840px or mobile device (**with space for updated nav component on the bottom)

Screen Shot 2022-07-26 at 4 26 15 PM
Screen Shot 2022-07-26 at 4 26 08 PM

### CODE REVIEW

File: src/pages/TokenDetails/index.tsx [hunk 1]
```diff
@@ -53,20 +69,69 @@ export default function TokenDetails() {
...
+    if (!userConnectedToATestNetwork) {
+      chainIds = chainIds.filter((id) => !(TESTNET_CHAIN_IDS as unknown as SupportedChainId[]).includes(id))
+    }
+    return chainIds
+  }, [connectedChainId])
+
+  const balancesByNetwork = data
+    ? chainsToList.map((chainId) => {
+        const amount = data[chainId]
+        console.log(`theme.chain_${chainId}`)
```
Comment (vm) [inline src/pages/TokenDetails/index.tsx:95]: rm

File: .vscode/settings.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "editor.formatOnSaveMode": "file",
   "editor.tabCompletion": "on",
   "editor.tabSize": 2,
-  "editor.formatOnSave": true,
+  "editor.formatOnSave": false,
```
Comment (vm) [inline .vscode/settings.json:8]: the bane of my existence

File: src/components/Explore/TokenDetails/BalanceSummary.tsx [hunk 3]
```diff
@@ -56,26 +53,24 @@ const TotalBalanceItem = styled.div`
...
 `
 
-export default function BalanceSummary({ address }: { address: string }) {
+export default function BalanceSummary({
+  address,
+  networkBalances,
+  totalBalance,
+}: {
+  address: string
+  networkBalances: (JSX.Element | null)[] | null
```
Comment (vm) [inline src/components/Explore/TokenDetails/BalanceSummary.tsx:62]: `networkBalances?: (JSX.Element | null)[]`

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  display: flex;
+  justify-content: flex-start;
+  flex-direction: column;
+`
+const FakeFooterNavBar = styled.div`
+  position: fixed;
+  bottom: 0px;
+  left: 0px;
+  background-color: ${({ theme }) => theme.backgroundBackdrop};
+  height: 56px;
```
Comment (vm) [inline src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:50]: is this related to https://github.com/Uniswap/interface/pull/4194/files#diff-a92c5018623f57e640137b5b68f3c081ef44909861486fb2102aee0dce534df7R23?

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  width: 100%;
+  align-items: flex-end;
+  padding: 20px 10px;
```
Comment (vm) [inline src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:53]: 8px

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+  font-size: 10px;
```
Comment (vm) [inline src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:54]: no 10px allowed

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+  align-items: center;
+  border: none;
+  color: ${({ theme }) => theme.textPrimary};
+  padding: 12px 16px;
+  width: 120px;
+  height: 44px;
+  font-size: 16px;
+  font-weight: 600;
+  justify-content: center;
+  cursor: pointer;
```
Comment (vm) [inline src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:91]: this isn't default on buttons?

File: src/pages/TokenDetails/index.tsx [hunk 8]
```diff
@@ -46,26 +60,81 @@ const Widget = styled.div`
...
+      chainIds = chainIds.filter((id) => !(TESTNET_CHAIN_IDS as unknown as SupportedChainId[]).includes(id))
+    }
+    return chainIds
+  }, [connectedChainId])
+
+  const balancesByNetwork = data
+    ? chainsToList.map((chainId) => {
+        const amount = data[chainId]
+        const fiatValue = amount // for testing purposes
+        if (!fiatValue || !isChainAllowed(connector, chainId)) return null
```
Comment (vm) [inline src/pages/TokenDetails/index.tsx:99]: why do you need to check if chainnotallowed?

Comment (vm) [reply src/pages/TokenDetails/index.tsx:99]: if it's not allowed the balance would just be 0 right? idk lmk what you think

File: src/pages/TokenDetails/index.tsx [hunk 9]
```diff
@@ -46,26 +60,81 @@ const Widget = styled.div`
...
+        const chainInfo = getChainInfo(chainId)
+        if (!chainInfo) return null
+        return (
+          <NetworkBalance
+            key={chainId}
+            logoUrl={chainInfo.logoUrl}
+            balance={'1'}
+            tokenSymbol={tokenSymbol ?? 'XXX'}
```
Comment (vm) [inline src/pages/TokenDetails/index.tsx:107]: defaulting seems bad, if we don't have a symbol we have bigger issues

File: src/pages/TokenDetails/index.tsx [hunk 10]
```diff
@@ -46,26 +60,81 @@ const Widget = styled.div`
...
+            fiatValue={fiatValue.toSignificant(2)}
+            label={chainInfo.label}
+            networkColor={theme.textPrimary}
```
Comment (vm) [inline src/pages/TokenDetails/index.tsx:110]: networkcolor is a textprimary?

File: src/pages/TokenDetails/index.tsx (hunk 9 above)
Comment (kayleegeorge) [reply src/pages/TokenDetails/index.tsx:107]: oh ya this was a placeholder haha, Fred recently gave me a token details state on Figma that I needa do

File: src/pages/TokenDetails/index.tsx (hunk 10 above)
Comment (kayleegeorge) [reply src/pages/TokenDetails/index.tsx:110]: HMMM this was correct before, i wonder if I merged and this got overwritten -- I'll go through everything again

File: src/pages/TokenDetails/index.tsx (hunk 8 above)
Comment (kayleegeorge) [reply src/pages/TokenDetails/index.tsx:99]: oh ya this was jordan i think

Comment (vm) [reply src/pages/TokenDetails/index.tsx:99]: alright can we remove the chain allowed checks?

File: .vscode/settings.json (hunk 2 above)
Comment (kayleegeorge) [reply .vscode/settings.json:8]: NO WAY THIS IS HERE HAHA

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx (hunk 4 above)
Comment (kayleegeorge) [reply src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:50]: ya exactly -- it's basically a placeholder for the amount of space the nav bar will end up taking up

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx (hunk 6 above)
Comment (kayleegeorge) [reply src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:54]: i think it's ok for font

File: src/components/Explore/TokenDetails/FooterBalanceSummary.tsx (hunk 4 above)
Comment (vm) [reply src/components/Explore/TokenDetails/FooterBalanceSummary.tsx:50]: if its related... use a variable

**Review by vm — APPROVED** [review]

**Pull request merged**